crates/sovereign-sdk/adapters/risc0/src/host.rs

```diff
@@ -92,18 +92,24 @@ impl<'a> ZkvmHost for Risc0Host<'a> {
 
     fn extract_output<Da: sov_rollup_interface::da::DaSpec, Root: BorshDeserialize>(
         proof: &Proof,
-    ) -> Result<sov_rollup_interface::zk::StateTransition<Da, Root>, Self::Error> {
-        let journal = match proof {
+    ) -> Result<
+        (
+            sov_rollup_interface::zk::StateTransition<Da, Root>,
+            Option<Receipt>,
+        ),
+        Self::Error,
+    > {
+        let (journal, receipt) = match proof {
             Proof::PublicInput(journal) => {
                 let journal: Journal = bincode::deserialize(journal)?;
-                journal
+                (journal, None)
             }
             Proof::Full(data) => {
                 let receipt: Receipt = bincode::deserialize(data)?;
-                receipt.journal
+                (receipt.journal.clone(), Some(receipt))
             }
         };
-        Ok(BorshDeserialize::deserialize(&mut journal.bytes.as_ref())?)
+        Ok((BorshDeserialize::try_from_slice(&journal.bytes)?, receipt))
     }
```

Comment:
this method just keeps getting uglier.

you want to add another method to the trait where it returns the receipt itself, not just the output?

Reply:
no longer relevant with the current version